spring-boot-project/spring-boot-actuator-autoconfigure/src/main/java/org/springframework/boot/actuate/autoconfigure/integration/IntegrationGraphEndpointAutoConfiguration.java

```diff
@@ -0,0 +1,49 @@
+/*
+ * Copyright 2012-2018 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.boot.actuate.autoconfigure.integration;
+
+import org.springframework.boot.actuate.autoconfigure.endpoint.condition.ConditionalOnEnabledEndpoint;
+import org.springframework.boot.actuate.integration.IntegrationGraphEndpoint;
+import org.springframework.boot.autoconfigure.EnableAutoConfiguration;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnBean;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnClass;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnMissingBean;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.integration.support.management.graph.IntegrationGraphServer;
+
+/**
+ * {@link EnableAutoConfiguration Auto-configuration} for the {@link IntegrationGraphEndpoint}.
+ *
+ * @author Tim Ysewyn
+ * @since 2.1.0
+ */
+@Configuration
+@ConditionalOnClass(IntegrationGraphServer.class)
+public class IntegrationGraphEndpointAutoConfiguration {
```

Comment:
You have to add this to `spring.factories` as well, see `/spring-boot-project/spring-boot-actuator-autoconfigure/src/main/resources/spring.factories`.

This mechanism facilitates the fact that the autoconfiguration classes can be looked up instead are discovered via a package scan.

Reply:
Good catch! For some reason it didn't get committed & pushed, but I do remember making the change 🤔
Anyway, will do that right away! Thanks!